core/Stream/hsStream.cpp

```diff
@@ -381,83 +430,128 @@ bool hsFileStream::open(const ST::string& file, FileMode mode)
     } else if (mode == fmRead || mode == fmReadWrite) {
         throw hsFileReadException(__FILE__, __LINE__, "File does not exist");
     }
+#endif
     return false;
 }
 
 void hsFileStream::close()
 {
-    if (F)
+    if (F) {
+#ifdef WIN32
+        CloseHandle(F);
+#else
         fclose(F);
+#endif
+    }
     F = nullptr;
 }
 
 uint32_t hsFileStream::size() const
 {
     if (F == nullptr)
         return 0;
+
+#ifdef WIN32
+    DWORD sz = GetFileSize(F, nullptr);
+#else
     unsigned int p = ftell(F);
     fseek(F, 0, SEEK_END);
     unsigned int sz = ftell(F);
     fseek(F, p, SEEK_SET);
+#endif
     return sz;
 }
 
 uint32_t hsFileStream::pos() const
 {
     if (F == nullptr)
         return 0;
+#ifdef WIN32
+    return SetFilePointer(F, 0, nullptr, FILE_CURRENT);
+#else
     return ftell(F);
+#endif
 }
 
 bool hsFileStream::eof() const
 {
     if (F == nullptr)
         return true;
+
+#ifdef WIN32
+    return pos() == size();
```

Comment:
Hmm, the Posix implementation worked even for files whose size was unknown...  Arguably, that may never be relevant to libhsplasma, but I wonder if there's a way to preserve those semantics here.